packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -82,28 +83,30 @@ const shouldApplyFilterItemOnGroup = (columnField: string, node: GridGroupNode)
 export const filterRowTreeFromGroupingColumns = (
   params: FilterRowTreeFromTreeDataParams,
 ): Omit<GridFilterState, 'filterModel'> => {
-  const { rowTree, isRowMatchingFilters, filterModel } = params;
+  const { apiRef, rowTree, isRowMatchingFilters, filterModel } = params;
   const filteredRowsLookup: Record<GridRowId, boolean> = {};
   const filteredDescendantCountLookup: Record<GridRowId, number> = {};
 
   const filterTreeNode = (
     node: GridTreeNode,
     areAncestorsExpanded: boolean,
-    ancestorsResults: ReturnType<GridAggregatedFilterItemApplier>[],
+    ancestorsResults: GridAggregatedFilterItemApplierResult[],
   ): number => {
-    let isPassingFiltering = false;
-    let filterResults: ReturnType<GridAggregatedFilterItemApplier> = {
+    const filterResults: GridAggregatedFilterItemApplierResult = {
       passingFilterItems: null,
       passingQuickFilterValues: null,
     };
 
+    let isPassingFiltering = false;
+
     if (isRowMatchingFilters && node.type !== 'footer') {
       const shouldApplyItem =
         node.type === 'group' && node.isAutoGenerated
           ? (columnField: string) => shouldApplyFilterItemOnGroup(columnField, node)
           : undefined;
 
-      filterResults = isRowMatchingFilters(node.id, shouldApplyItem);
+      const row = apiRef.current.getRow(node.id);
+      isRowMatchingFilters(row, shouldApplyItem, filterResults);
```

Comment:
What is the benefit of passing the `filterResults` to `isRowMatchingFilters` instead of returning it from `isRowMatchingFilters`?

Reply:
In the flat filtering case, we can reuse the same `filterResults` object instead of allocating a new one on each iteration.

Reply:
Right, I didn't look into flat filtering 👍